e2e/cypress/integration/messaging/inline_images_open_preview_window_spec.js

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @messaging
+
+describe('Messaging', () => {
+    before(() => {
+        // # Login as test user and visit the newly created test channel
+        cy.apiInitSetup({loginAfter: true}).then(({team, channel}) => {
+            // # Visit a test channel
+            cy.visit(`/${team.name}/channels/${channel.name}`);
+        });
+    });
+    it('MM-T187 Inline markdown images open preview window', () => {
+        // # Enable 'Show markdown preview option in message input box' setting in Account Settings > Advanced
+        cy.findAllByLabelText('main menu').should('be.visible').click();
+        cy.findByText('Account Settings').click();
+        cy.get('#accountSettingsModal').should('be.visible').within(() => {
+            cy.findByText('Advanced').click();
+            cy.findByText('Preview Pre-release Features').should('be.visible').click();
+            cy.get('#advancedPreviewFeaturesmarkdown_preview').check().should('be.checked');
+            cy.findByText('Save').click();
+            cy.findAllByLabelText('Close').first().click();
+        });
+
+        // # Post the image link to the channel
+        cy.postMessage('![test image](https://www.mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)');
+
+        // # Hover over image then click to open preview image
+        cy.get('.file-preview__button').trigger('mouseover').click();
+
+        // * Confirm image container is visible
+        cy.get('.markdown-inline-img__container').should('be.visible');
```

Comment:
I've got a failing test here where the image review is not showing. Looks like Cypress is clicking the preview button even if the button is not ready yet.
<img width="615" alt="Screen Shot 2020-09-28 at 5 50 41 AM" src="https://user-images.githubusercontent.com/5334504/94376820-a0fe5a00-014f-11eb-8ca2-3a7763f9f053.png">

The `markdown-inline-img__container` seemed to be checked before clicking file preview button.

Though not mentioned in the test case, I'd suggests the ff:
- add message text ("Hello") and wait until it's visibly posted to buy some time for the image to be loaded.
- check for `markdown-inline-img--hover` to fulfill "Drop shadow hover effect". This is brittle and should check for actual box-shadow property but there seemed to be no way in Cypress (feel free to play around in case I'm missing something)
```suggestion
        // # Post the image link to the channel
        cy.postMessage('Hello ![test image](https://www.mattermost.org/wp-content/uploads/2016/03/logoHorizontal.png)');

        // * Confirm image container is visible
        cy.uiWaitUntilMessagePostedIncludes('Hello');
        cy.get('.markdown-inline-img__container').should('be.visible');

        // # Hover over image then click to open preview image
        cy.findByAltText('test image').should('be.visible').and('have.class', 'markdown-inline-img--hover');
        cy.get('.file-preview__button').should('be.visible').trigger('mouseover').click();
```

Reply:
Thank-you - changes made! Not sure  how I reversed those two.